src/lamoda/routers.py

```diff
@@ -1,33 +1,144 @@
-from decimal import Decimal
+import json
 from typing import Annotated
 
-from bson import Decimal128
-from database import Collection, get_lamoda_collection
+from bson import ObjectId
 from fastapi import APIRouter, Depends
 
-from .service import parse_object
+from brokers.producer import producer
+from cache import RedisCacheManager
+from core.enums import ObjectStatus
+from db import get_lamoda_database
+from db.database_managers import LamodaDatabaseManager
+from dependecies import get_cache_manager
+from lamoda.config import LamodaSettings
+from lamoda.service import (
+    parse_lamoda_category,
+    parse_links_from_category,
+    parse_object,
+    validate_url,
+)
+from schemas import LamodaResponseFromParser
 
-lamoda_router = APIRouter(prefix='/lamoda')
+lamoda_router = APIRouter(prefix="/lamoda")
 
-LamodaCollection = Annotated[Collection, Depends(get_lamoda_collection)]
+LamodaDb = Annotated[LamodaDatabaseManager, Depends(get_lamoda_database)]
+CacheMngr = Annotated[RedisCacheManager, Depends(get_cache_manager)]
+settings = LamodaSettings()
 
 
 @lamoda_router.post("/product")
-def parse_product(url: str, collection: LamodaCollection):
-    """View for parsing product by its url"""
+async def parse_product(url: str, db: LamodaDb, cache: CacheMngr):
+    """View for parsing product by its url, processed product is saved to cache and db"""
 
+    validate_url(url, is_product=True, is_category=False)
```

Comment:
can you use
`async def parse_product(url = Depends(validate_url)`?

Reply:
same below

Reply:
I dont think it is possible to use it here, but i think that it is possible to make pydantic model so it will validate url there instantly when it gets into view